include/core/Ref.hpp

```diff
@@ -4,6 +4,7 @@
 #include "Variant.hpp"
 #include "GodotGlobal.hpp"
 #include "../Reference.hpp"
+#include "../Object.hpp"
```

Comment:
I think this was removed by @Zylann on purpose, I'm not sure if it's good to add it again

Reply:
This is what I get with out the `Object.hpp` file included

```
In file included from src/Reference.cpp:1:
In file included from include/core/../Reference.hpp:9:
include/core/Ref.hpp:95:3: error: use of undeclared identifier 'Reference'; did you mean 'reference'?
                Reference *refb = const_cast<Reference *>(static_cast<const Reference *>(p_from.ptr()));
                ^~~~~~~~~
                reference
include/core/Ref.hpp:15:5: note: 'reference' declared here
        T *reference = NULL;
           ^
include/core/Ref.hpp:95:32: error: unknown type name 'Reference'
                Reference *refb = const_cast<Reference *>(static_cast<const Reference *>(p_from.ptr()));
                                             ^
include/core/Ref.hpp:95:63: error: unknown type name 'Reference'
                Reference *refb = const_cast<Reference *>(static_cast<const Reference *>(p_from.ptr()));
                                                                            ^
include/core/Ref.hpp:95:14: error: use of undeclared identifier 'refb'
                Reference *refb = const_cast<Reference *>(static_cast<const Reference *>(p_from.ptr()));
                           ^
include/core/Ref.hpp:96:8: error: use of undeclared identifier 'refb'
                if (!refb) {
                     ^
include/core/Ref.hpp:102:21: error: use of undeclared identifier 'refb'
                r.reference = (T*)refb;
                                  ^
include/core/Ref.hpp:110:3: error: use of undeclared identifier 'Reference'; did you mean 'reference'?
                Reference *refb = (Reference *) (Object *) p_variant;
                ^~~~~~~~~
                reference
include/core/Ref.hpp:15:5: note: 'reference' declared here
        T *reference = NULL;
           ^
include/core/Ref.hpp:110:22: error: use of undeclared identifier 'Reference'
                Reference *refb = (Reference *) (Object *) p_variant;
                                   ^
include/core/Ref.hpp:110:33: error: expected expression
                Reference *refb = (Reference *) (Object *) p_variant;
                                              ^
include/core/Ref.hpp:110:36: error: 'Object' does not refer to a value
                Reference *refb = (Reference *) (Object *) p_variant;
                                                 ^
include/core/RID.hpp:8:7: note: declared here
class Object;
      ^
In file included from src/Reference.cpp:1:
In file included from include/core/../Reference.hpp:9:
include/core/Ref.hpp:110:44: error: expected expression
                Reference *refb = (Reference *) (Object *) p_variant;
                                                         ^
include/core/Ref.hpp:110:14: error: use of undeclared identifier 'refb'
                Reference *refb = (Reference *) (Object *) p_variant;
                           ^
include/core/Ref.hpp:111:8: error: use of undeclared identifier 'refb'
                if (!refb) {
                     ^
include/core/Ref.hpp:118:22: error: use of undeclared identifier 'refb'
                r.reference = (T *)refb;
                                   ^
include/core/Ref.hpp:134:3: error: use of undeclared identifier 'Reference'; did you mean 'reference'?
                Reference *refb = const_cast<Reference *>(static_cast<const Reference *>(p_from.ptr()));
                ^~~~~~~~~
                reference
include/core/Ref.hpp:15:5: note: 'reference' declared here
        T *reference = NULL;
           ^
include/core/Ref.hpp:134:32: error: unknown type name 'Reference'
                Reference *refb = const_cast<Reference *>(static_cast<const Reference *>(p_from.ptr()));
                                             ^
include/core/Ref.hpp:134:63: error: unknown type name 'Reference'
                Reference *refb = const_cast<Reference *>(static_cast<const Reference *>(p_from.ptr()));
                                                                            ^
include/core/Ref.hpp:134:14: error: use of undeclared identifier 'refb'
                Reference *refb = const_cast<Reference *>(static_cast<const Reference *>(p_from.ptr()));
                           ^
include/core/Ref.hpp:135:8: error: use of undeclared identifier 'refb'
                if (!refb) {
```

Reply:
Hmmm okay, @Zylann should take a second look :)